src/sgame/sg_bot_util.cpp

```diff
@@ -317,35 +317,39 @@ void BotGetDesiredBuy( gentity_t *self, weapon_t *weapon, upgrade_t *upgrades, i
 	int credits = self->client->ps.persistant[PERS_CREDIT];
 	int usableCapital = credits + equipmentPrice;
 
+	//could not use inline function since using "too_expensive" value (usableCapital) inside.
```

Comment:
You could use a lambda function. `auto CanBuy = [usableCapital](weapon_t w) { return ...; };`